playbooks/datadog/tag-slack-users.yml

```diff
@@ -0,0 +1,106 @@
+title: Send Slack notifications for Datadog SIEM signals
+description: |
+  Pulls Datadog SIEM signals (alerts), extracts emails contains within the signal,
+  finds Slack users associated with the emails, and tags the Slack users in a notification.
+entrypoint: pull_datadog_siem_signals
+inputs:
+  dd_region: us5
+  dd_api_url: https://api.us5.datadoghq.com
+  slack_channel: CXXXXXXXXXXXX
+  slack_webhook: http://your-slack-webhook-url.com
+
+triggers:
+  - type: webhook
+    ref: my_webhook
+    entrypoint: pull_datadog_siem_signals
+
+actions:
+  - ref: pull_datadog_siem_signals
+    action: integrations.datadog.siem.list_datadog_alerts
+    args:
+      start_time: ${{ TRIGGER.start_time }}
+      end_time: ${{ TRIGGER.end_time }}
+      limit: 10
+
+  - ref: reshape_signals
+    action: core.transform.forward
+    depends_on:
+      - pull_datadog_siem_signals
+    run_if: ${{ FN.not_empty(ACTIONS.pull_datadog_siem_signals.result) }}
+    for_each: ${{ for var.signal in ACTIONS.pull_datadog_siem_signals.result }}
+    args:
+      value:
+        title: ${{ var.signal.attributes.attributes.title }}
+        description: ${{ var.signal.attributes.message }}
+        status: ${{ var.signal.attributes.status }}
+        first_seen: ${{ var.signal.attributes.attributes.workflow.first_seen }}
+        last_seen: ${{ var.signal.attributes.attributes.workflow.last_seen }}
+        log_samples: ${{ FN. }}
+        link_to_signal: ${{ INPUTS.dd_api_url }}/security?event=${{ var.signal.EventID }}
+
+  - ref: extract_emails
+    action: integrations.extraction.extract_emails
+    depends_on:
+      - reshape_signals
+    for_each: ${{ for var.signal in ACTIONS.reshape_signals.result }}
+    args:
+      text: ${{ var.signal.log_samples }}
+
+  - ref: find_slack_users
+    action: integrations.slack.find_slack_users
+    depends_on:
+      - extract_emails
+    for_each: ${{ for var.emails in ACTIONS.extract_emails.result }}
+    args:
+      emails: ${{ var.emails }}
+
+  - ref: extract_slack_user_ids
+    action: core.transform.forward
+    depends_on:
+      - find_slack_users
+    for_each: ${{ for var.users in ACTIONS.find_slack_users.result }}
+    args:
+      # TODO: Implement nested for loops nd ability to apply function to each member...
+      ids: ${{ for <@{{ var.user.id }}> for var.user in var.users}}
```

Comment:
@daryllimyt please assist in better API designs (more detailed comments in Slack DM)

Reply:
Resolved by #172